File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/CodeSystem.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class CodeSystem : DomainResource, IResource
{
    public string? Description { get; set; }
    public string? Date { get; set; }
    public bool? VersionNeeded { get; set; }
    public string? Publisher { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public string? Purpose { get; set; }
    public required string Content { get; set; }
    public CodeSystem_Property[]? Property { get; set; }
    public string? Name { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public string? Copyright { get; set; }
    public bool? Experimental { get; set; }
    public string? Title { get; set; }
    public CodeSystem_Filter[]? Filter { get; set; }
    public string? Supplements { get; set; }
    public bool? Compositional { get; set; }
    public required string Status { get; set; }
    public string? HierarchyMeaning { get; set; }
    public string? ValueSet { get; set; }
    public uint? Count { get; set; }
    public string? Url { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public CodeSystem_Concept[]? Concept { get; set; }
    public bool? CaseSensitive { get; set; }
    public string? Version { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }

    public class CodeSystem_Property : BackboneElement
    {
        public string? Uri { get; set; }
        public required string Code { get; set; }
        public required string Type { get; set; }
        public string? Description { get; set; }
    }

    public class CodeSystem_Filter : BackboneElement
    {
        public required string Code { get; set; }
        public required string Value { get; set; }
        public required string[] Operator { get; set; }
        public string? Description { get; set; }
    }

    public class CodeSystem_Concept_Concept_Property : BackboneElement
    {
        public string? ValueCode { get; set; }
        public string? ValueDecimal { get; set; }
        public object? Value    
        {
            get
            {
                if (ValueCode is not null)
                {
                    return ValueCode;
                }
        
                if (ValueDecimal is not null)
                {
                    return ValueDecimal;
                }
        
                if (ValueString is not null)
                {
                    return ValueString;
                }
        
                if (ValueBoolean is not null)
                {
                    return ValueBoolean;
                }
        
                if (ValueDateTime is not null)
                {
                    return ValueDateTime;
                }
        
                if (ValueCoding is not null)
                {
                    return ValueCoding;
                }
        
                if (ValueInteger is not null)
                {
                    return ValueInteger;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    ValueCode = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDecimal = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueString = (string)value;return;
                }
        
                if (value?.GetType() == typeof(bool))
                {
                    ValueBoolean = (bool)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDateTime = (string)value;return;
                }
        
                if (value?.GetType() == typeof(Base.Coding))
                {
                    ValueCoding = (Base.Coding)value;return;
                }
        
                if (value?.GetType() == typeof(int))
                {
                    ValueInteger = (int)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public string? ValueString { get; set; }
        public bool? ValueBoolean { get; set; }
        public string? ValueDateTime { get; set; }
        public Base.Coding? ValueCoding { get; set; }
        public required string Code { get; set; }
        public int? ValueInteger { get; set; }
    }

    public class CodeSystem_Concept_Concept_Designation : BackboneElement
    {
        public Base.Coding? Use { get; set; }
        public required string Value { get; set; }
        public string? Language { get; set; }
    }

    public class CodeSystem_Concept_Concept : BackboneElement
    {
        public required string Code { get; set; }
        public string[]? Concept { get; set; }
        public string? Display { get; set; }
        public CodeSystem_Concept_Concept_Property[]? Property { get; set; }
        public string? Definition { get; set; }
        public CodeSystem_Concept_Concept_Designation[]? Designation { get; set; }
    }

    public class CodeSystem_Concept_Property : BackboneElement
    {
        public string? ValueCode { get; set; }
        public string? ValueDecimal { get; set; }
        public object? Value    
        {
            get
            {
                if (ValueCode is not null)
                {
                    return ValueCode;
                }
        
                if (ValueDecimal is not null)
                {
                    return ValueDecimal;
                }
        
                if (ValueString is not null)
                {
                    return ValueString;
                }
        
                if (ValueBoolean is not null)
                {
                    return ValueBoolean;
                }
        
                if (ValueDateTime is not null)
                {
                    return ValueDateTime;
                }
        
                if (ValueCoding is not null)
                {
                    return ValueCoding;
                }
        
                if (ValueInteger is not null)
                {
                    return ValueInteger;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    ValueCode = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDecimal = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueString = (string)value;return;
                }
        
                if (value?.GetType() == typeof(bool))
                {
                    ValueBoolean = (bool)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDateTime = (string)value;return;
                }
        
                if (value?.GetType() == typeof(Base.Coding))
                {
                    ValueCoding = (Base.Coding)value;return;
                }
        
                if (value?.GetType() == typeof(int))
                {
                    ValueInteger = (int)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public string? ValueString { get; set; }
        public bool? ValueBoolean { get; set; }
        public string? ValueDateTime { get; set; }
        public Base.Coding? ValueCoding { get; set; }
        public required string Code { get; set; }
        public int? ValueInteger { get; set; }
    }

    public class CodeSystem_Concept_Designation : BackboneElement
    {
        public Base.Coding? Use { get; set; }
        public required string Value { get; set; }
        public string? Language { get; set; }
    }

    public class CodeSystem_Concept : BackboneElement
    {
        public required string Code { get; set; }
        public CodeSystem_Concept_Concept[]? Concept { get; set; }
        public string? Display { get; set; }
        public CodeSystem_Concept_Property[]? Property { get; set; }
        public string? Definition { get; set; }
        public CodeSystem_Concept_Designation[]? Designation { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/CodeSystem.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import UsageContext
from base import ContactDetail
from base import CodeableConcept
from base import Coding
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class CodeSystemProperty(BackboneElement):
    uri: Optional[str] = None
    code: str
    type: str
    description: Optional[str] = None

@dataclass(kw_only=True)
class CodeSystemFilter(BackboneElement):
    code: str
    value: str
    operator: list[str] = field(default_factory=list)
    description: Optional[str] = None

@dataclass(kw_only=True)
class CodeSystemConceptProperty(BackboneElement):
    value_code: Optional[str] = None
    value_decimal: Optional[float] = None
    value_string: Optional[str] = None
    value_boolean: Optional[bool] = None
    value_date_time: Optional[str] = None
    value_coding: Optional[Coding] = None
    code: str
    value_integer: Optional[int] = None

@dataclass(kw_only=True)
class CodeSystemConceptDesignation(BackboneElement):
    use: Optional[Coding] = None
    value: str
    language: Optional[str] = None

@dataclass(kw_only=True)
class CodeSystemConcept(BackboneElement):
    code: str
    concept: Optional[List[Reference]] = field(default_factory=list)
    display: Optional[str] = None
    property: Optional[List[CodeSystemConceptProperty]] = field(default_factory=list)
    definition: Optional[str] = None
    designation: Optional[List[CodeSystemConceptDesignation]] = field(default_factory=list)

@dataclass(kw_only=True)
class CodeSystem(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    version_needed: Optional[bool] = None
    publisher: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = field(default_factory=list)
    purpose: Optional[str] = None
    content: str
    property: Optional[List[CodeSystemProperty]] = field(default_factory=list)
    name: Optional[str] = None
    use_context: Optional[List[UsageContext]] = field(default_factory=list)
    copyright: Optional[str] = None
    experimental: Optional[bool] = None
    title: Optional[str] = None
    filter: Optional[List[CodeSystemFilter]] = field(default_factory=list)
    supplements: Optional[str] = None
    compositional: Optional[bool] = None
    status: str
    hierarchy_meaning: Optional[str] = None
    value_set: Optional[str] = None
    count: Optional[int] = None
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    concept: Optional[List[CodeSystemConcept]] = field(default_factory=list)
    case_sensitive: Optional[bool] = None
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = field(default_factory=list)